variable/test/variable_test.cpp

```diff
@@ -26,6 +26,15 @@ TEST(Variable, construct) {
   EXPECT_EQ(data.size(), 2);
 }
 
+TEST(Variable, construct_boost_units_quantity) {
+  using boost::units::quantity;
+  EXPECT_EQ(Variable(quantity{1.2 * boost::units::si::meter}),
+            makeVariable<double>(Values{1.2}, units::Unit(units::m)));
+  EXPECT_EQ(Variable(quantity<boost::units::si::length, float>{
+                1.2 * boost::units::si::meter}),
+            makeVariable<float>(Values{1.2}, units::Unit(units::m)));
+}
+
```

Comment:
I see a number of tests for `float` and `double` in this file but no sanity checks for other types. Should we add simple tests for `Variable`s containing, say `bool`s

Reply:
If you are talking about the `makeVariable`, those are tested in more detail in https://github.com/scipp/scipp/blob/master/variable/test/variable_keyword_args_constructor_test.cpp.

Generally speaking you are right that `variable_test.cpp` can do with a cleanup (it is very old, with many tests from very early versions of the scipp prototype). We are actually planning on doing more overhaul of our tests since this is not the only place that could do with better tests. That being said, missing tests for different types are imho not the problem since everything is either type-erase or templated (and if that mechanism wouldn't work 90% of our unit tests would fail).